### PR TITLE
feat(workspace): Show notice for manual migration if upgrading from from legacy versions

### PR DESCRIPTION
**Pull request by hikchoi**
# feat(workspace): Show notice for manual migration if upgrading from from legacy versions

This PR:
- Removes pre-0.63 migration logics
- Adds prompt for manual migration for users who upgraded from pre-0.63 dendron.

# Pull Request Checklist

# Dendron Extended PR Checklist

- NOTE: the links don't work. you'll need to go into the wiki and use lookup to find the note until we fix some issues in the markdown export

### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [x] sticking to existing conventions instead of creating new ones
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended
- General
  - [x] check whether code be simplified
  - [x] check if similar function already exist in the codebase. if so, can it be re-used?
  - [x] check if this change adversely impact performance
- Operations
  - [x] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [x] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics
- [x] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### Extended
- [x] can we track the performance of this change to know if it is _successful_?
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

### Basics

- [x] [Write Tests](dev.process.qa.test) 
- [x] [Confirm existing tests pass](dev.process.qa.test)
- [x] [Confirm manual testing](dev.process.qa.test) 
- [x] Common cases tested
- [x] 1-2 Edge cases tested
- [x] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

### Extended
- [~] If you are adding a new language feature (graphically visible in VS Code/preview/publishing), an example is included in the [test workspace](dev.ref.test-workspace)
- CSS
  - [~] display is correct for following dimensions
    - [~] sm: screen ≥ 576px, eg. iphonex, (375x812)
    - [~] lg: screen ≥ 992px
    - [~] xxl: screen ≥ 1600px eg. mac (1600x900)
  - [~] display is correct for following browsers (across the various dimensions)
    - [~] safari
    - [~] firefox
    - [~] chrome

## Docs
- [~] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
- [~] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](dev) or [Packages](pkg)

### Extended
- [~]  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](dev.process.close-loop)
  - eg: [breaking dev change due to new dependency](https://github.com/dendronhq/dendron/pull/2188#pullrequestreview-855696330)

### CODE REVIEW

File: packages/plugin-core/src/utils/StartupUtils.ts [hunk 1]
```diff
@@ -35,8 +35,115 @@ import { SurveyUtils } from "../survey";
...
+    return (
+      workspaceInstallStatus === InstallStatus.UPGRADED &&
+      semver.lte(previousWorkspaceVersion, "0.63.0")
+    );
+  }
+
+  static showManualUpgradeMessage() {
+    const SHOW_ME_HOW = "Show Me How";
+    const MESSAGE =
+      "You are upgrading from a really old version of Dendron. Please follow the instructions to manually migrate your configurations.";
```
Comment (kevinslin) [inline packages/plugin-core/src/utils/StartupUtils.ts:62]: tweak this:

>       "You are upgrading from a legacy version of Dendron. Please follow the instructions below to manually migrate your configuration.";

File: packages/plugin-core/src/utils/StartupUtils.ts [hunk 2]
```diff
@@ -35,8 +35,115 @@ import { SurveyUtils } from "../survey";
...
+          AnalyticsUtils.track(MigrationEvents.ManualUpgradeMessageConfirm, {
+            status: ConfirmStatus.accepted,
+          });
+
+          const content = [
+            "# Instructions for Upgrade",
+            "",
+            "You are upgrading from a version that is below 0.63.0, which we do not support direct automatic migration.",
+            "",
+            "You can either perform a stepped upgrade, or manually move the migrated configurations to the correct location.",
```
Comment (kevinslin) [inline packages/plugin-core/src/utils/StartupUtils.ts:76]: I don't think most folks will be familiar with what a "stepped upgrade" means. would rephrase - you can either manually update the configuration or update to a compatible older version before upgrading to the current version

also - since these instructions can change and have multiple steps, would be better to link to the dendron wiki vs trying to write it out in this markdown message.

**Review by kevinslin — CHANGES_REQUESTED** [review]
heads up theres an issue with version numbers in master for native workspaces. can you rebase off of this to fix the test issues?

https://github.com/dendronhq/dendron/pull/3168

File: packages/plugin-core/src/utils/StartupUtils.ts (hunk 2 above)
Comment (hikchoi) [reply packages/plugin-core/src/utils/StartupUtils.ts:76]: If we are going to link to a note in the wiki, should we just ditch the webview altogether and just open the link when the toast message was accepted?

Comment (kevinslin) [reply packages/plugin-core/src/utils/StartupUtils.ts:76]: yep

**Comment by hikchoi** [comment]
> Should the last part of the title be 'manual migration for legacy users' or 'manual migration from legacy versions'?

Made the title more clear: `feat(workspace): Show notice for manual migration if upgrading from from legacy versions`

**Review by kevinslin — APPROVED** [review]
